src/libraries/System.Private.CoreLib/src/System/Threading/ThreadPool.cs

```diff
@@ -1070,6 +1067,19 @@ bool executeOnlyOnce
             return RegisterWaitForSingleObject(waitObject, callBack, state, (uint)tm, executeOnlyOnce, false);
         }
 
+        // The thread pool maintains a per-appdomain managed work queue.
+        // New thread pool entries are added in the managed queue.
+        // The VM is responsible for the actual growing/shrinking of
+        // threads.
+        private static void EnsureInitialized()
+        {
+            // Inspecting the readonly static ThreadPoolInitialized should initialize the ThreadPool
+            if (!ThreadPoolGlobals.ThreadPoolInitialized)
+            {
+                ThrowHelper.ThrowInvalidOperationException_ThreadPoolNotInitialized();
```

Comment:
In what situation can this happen?

Reply:
Was mostly something to put in the `if` block 😅